src/main/java/org/zeromq/ZMQ.java

```diff
@@ -1526,13 +1508,11 @@ public Poller(int size)
          */
         protected Poller(Context context, int size)
         {
-            if (context != null) {
-                this.context = context;
-            }
-
-            items = new PollItem[size];
-            timeout = -1L;
-            next = 0;
+            this.context = context;
+            selector     = context.selector();
+            items        = new PollItem[size];
```

Comment:
Add assert(selector != null) would be nice checking and fit with the current code